lib/mux.ml

```diff
@@ -71,6 +70,7 @@ end = struct
     transport : R.transport;
     outgoing_mutex: Lwt_mutex.t;
     id_to_wakeup : (R.id, R.request_hdr * ((unit, Protocol.Error.t) result Lwt.u) * R.response_body) Hashtbl.t;
+          * (unit, Protocol.Error.t) result Lwt.u
```

Comment:
Does this even compile..?

Reply:
The mux tests are compiled but don't seem to run. I've changed the setup so they get run... and they compile just fine, strangely.